lib/githubbot.ts

```diff
@@ -0,0 +1,403 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as ProcBot from './procbot';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import { AlertLevel } from './procbot';
+
+// The GithubAPI Promise implies built-in ES6 Promises. Therefore when assigning
+// the Promise to use as Bluebird, VS Code reports an error. Need a good
+// fix for this that doesn't require type declarations?
+const GithubApi = require('github');
```

Comment:
So to be clearer, there are `@type` declarations for `github`, but importing with:
```
import * as GithubApi from 'github';
```
 fails. This, I think, is due to the constructor itself takes a 'Promise' property (see line 204 of `githubbot.ts`):
```
lib/githubbot.ts(198,40): error TS2345: Argument of type '{ protocol: string; host: string; headers: { 'Accept': string; }; Promise: typeof Bluebird; timeo...' is not assignable to parameter of type 'Options | undefined'.
  Type '{ protocol: string; host: string; headers: { 'Accept': string; }; Promise: typeof Bluebird; timeo...' is not assignable to type 'Options'.
    Types of property 'Promise' are incompatible.
      Type 'typeof Bluebird' is not assignable to type 'PromiseConstructor | undefined'.
        Type 'typeof Bluebird' is not assignable to type 'PromiseConstructor'.
          Types of property 'all' are incompatible.
            Type '{ <T1, T2, T3, T4, T5>(values: [T1 | Thenable<T1>, T2 | Thenable<T2>, T3 | Thenable<T3>, T4 | The...' is not assignable to type '{ <TAll>(values: Iterable<TAll | PromiseLike<TAll>>): Promise<TAll[]>; <T1, T2, T3, T4, T5, T6, T...'.
              Type 'Bluebird<[any, any, any, any, any]>' is not assignable to type 'Promise<any[]>'.
                Types of property 'then' are incompatible.
                  Type '{ <U1, U2>(onFulfill: (value: [any, any, any, any, any]) => U1 | Thenable<U1>, onReject: (error: ...' is not assignable to type '{ (onfulfilled?: ((value: any[]) => any[] | PromiseLike<any[]>) | null | undefined, onrejected?: ...'.
                    Type 'Bluebird<any>' is not assignable to type 'Promise<any[]>'.
                      Types of property 'then' are incompatible.
                        Type '{ <U1, U2>(onFulfill: (value: any) => U1 | Thenable<U1>, onReject: (error: any) => U2 | Thenable<...' is not assignable to type '{ (onfulfilled?: ((value: any[]) => any[] | PromiseLike<any[]>) | null | undefined, onrejected?: ...'.
                          Type 'Bluebird<any>' is not assignable to type 'Promise<any[]>'.
```